core-web/libs/edit-content/src/lib/fields/dot-edit-content-custom-field/dot-edit-content-custom-field.component.ts

```diff
@@ -31,58 +33,127 @@ import { WINDOW } from '@dotcms/utils';
             provide: WINDOW,
             useValue: window
         }
+    ],
+    viewProviders: [
+        {
+            provide: ControlContainer,
+            useFactory: () => inject(ControlContainer, { skipSelf: true })
+        }
     ]
 })
-export class DotEditContentCustomFieldComponent implements OnInit {
-    @Input() field!: DotCMSContentTypeField;
-    @Input() contentType!: string;
+export class DotEditContentCustomFieldComponent implements OnDestroy {
+    $field = input<DotCMSContentTypeField>(null, { alias: 'field' });
+    $contentType = input<string>(null, { alias: 'contentType' });
+    iframe = viewChild.required<ElementRef<HTMLIFrameElement>>('iframe');
 
-    @ViewChild('iframe') iframe!: ElementRef<HTMLIFrameElement>;
+    #window = inject(WINDOW);
+    private readonly ALLOWED_ORIGINS = [this.#window.location.origin];
 
-    private controlContainer = inject(ControlContainer);
-    private zone = inject(NgZone);
+    $isFullscreen = signal(false);
+    $variables = signal<Record<string, string>>({});
 
-    variables!: { [key: string]: string };
-    isFullscreen = false;
-    src!: string;
+    $src = computed(() => {
+        const field = this.$field();
+        const contentType = this.$contentType();
 
-    constructor(@Inject(WINDOW) private window: Window) {}
+        if (!field || !contentType) {
+            return '';
+        }
 
-    ngOnInit() {
-        this.src = `/html/legacy_custom_field/legacy-custom-field.jsp?variable=${this.contentType}&field=${this.field.variable}`;
-        this.variables = this.field.fieldVariables.reduce((result, item) => {
-            result[item.key] = item.value;
+        const params = new URLSearchParams({
+            variable: contentType,
+            field: field.variable
+        });
 
-            return result;
-        }, {});
-    }
+        return `/html/legacy_custom_field/legacy-custom-field.jsp?${params}`;
+    });
+
+    #formBridge?: FormBridge;
```

Comment:
@oidacra Why is it optional? I think it should have null as an initial value.

Reply:
Fixed. It will be null.